encryption/encryption.cc

```diff
@@ -536,6 +536,14 @@ Status KeyManagedEncryptedEnv::RenameFile(const std::string& src_fname,
   return s;
 }
 
+Status KeyManagedEncryptedEnv::DeleteDir(const std::string& dname) {
+  Status s = target()->DeleteDir(dname);
+  if (!s.ok()) {
+    return s;
+  }
+  return key_manager_->DeleteFile(dname);
```

Comment:
How about also give it a `DeleteDir` method? It's confusing to mix use `DeleteFile` and `DeleteDir`.

Reply:
But it's already mixed used in `Env::RenameFile`. We can't know if RocksDB is renaming a dir or file there.